src/content/extension.js

```diff
@@ -1,24 +1,261 @@
-import GitHub from "./github";
-import LeetCode from "./leetCode";
-import StackOverflow from "./stackOverflow";
+/* globals chrome */
 
+import prettier from "prettier/standalone";
 import Storage from "./storage";
+import { PARSERS, PARSERS_LANG_MAP } from "./parsers";
 
-const extensions = [GitHub, StackOverflow, LeetCode];
+const STACKEXCHANGE_SITES = [
+  "https://stackoverflow.com",
+  "https://askubuntu.com",
+  "https://mathoverflow.com",
+  "https://serverfault.com",
+  "https://stackapps.com",
+  "https://superuser.com",
+];
+const STACKEXCHANGE_URL_REGEX = /^https:\/\/([a-z]+).stackexchange.com/;
+const STACKEXCHANGE_VALID_PATHNAMES = /(^\/questions|\/posts\/\d+\/edit|^\/review)/u;
 
-export default class {
+export default class Extension {
   constructor() {
     this._storage = new Storage();
   }
 
   async init() {
     await this._storage.init();
+    this._bindListeners();
+  }
+
+  _bindListeners() {
+    chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
+      if (message.action !== "runPrettierFormat") {
+        return;
+      }
+
+      const inputEl = document.activeElement;
+
+      if (!inputEl) {
```

Comment:
document.activeElement will always exist. we need filter it for supported input elements. like only textarea.

for now we can do.

```suggestion
      if (inputEl?.tagName === 'TEXTAREA') {
```

I don't think text input boxes needs to be supported as no one is going to use it for code.